src/dbnode/storage/shard.go

```diff
@@ -1992,10 +2003,12 @@ func (s *dbShard) ColdFlush(
 	// a block, we continue to try persisting other blocks.
 	for blockStart := range dirtySeriesToWrite {
 		startTime := blockStart.ToTime()
+		coldVersion := s.RetrievableBlockColdVersion(startTime)
 		fsID := fs.FileSetFileIdentifier{
-			Namespace:  s.namespace.ID(),
-			Shard:      s.ID(),
-			BlockStart: startTime,
+			Namespace:   s.namespace.ID(),
+			Shard:       s.ID(),
+			BlockStart:  startTime,
+			VolumeIndex: coldVersion,
```

Comment:
The call to `Merge` will error out if our logic is wrong and the volume does not exist right? (which is what we want it to do)

Reply:
Yep, it will fail to open the file and error out.